website/docs/core-concepts/validate/terraform-variables.mdx

```diff
@@ -0,0 +1,196 @@
+    # Supports both absolute and relative paths
+    base_path: "stacks/schemas/opa"
+```
+</File>
+
+In the component manifest, add the `settings.validation` section to point to the OPA policy file:
+
+<File title="stack.yaml">
+```yaml
+components:
+  terraform:
+    my-component:
+      settings:
+        # All validation steps must succeed to allow the component to be provisioned
+        validation:
+          check-template-functions-test-component-with-opa-policy:
+            schema_type: opa
+            # 'schema_path' can be an absolute path or a path relative to 'schemas.opa.base_path' defined in `atmos.yaml`
+            schema_path: "my-component/validate-my-component.rego"
+            description: Check 'my-component' component using OPA policy
+            # Validation timeout in seconds
+            timeout: 5
+```
+</File>
+
+### Require a Terraform variable to be specified on the command line
+
+If you need to enforce that a Terraform variable must be specified on the command line (and not in Atmos stack manifests),
+add the following OPA policy in the file `stacks/schemas/opa/my-component/validate-my-component.rego`
+
+<File title="stacks/schemas/opa/my-component/validate-my-component.rego">
+```rego
+# 'package atmos' is required in all `atmos` OPA policies
+package atmos
+
+# Atmos looks for the 'errors' (array of strings) output from all OPA policies.
+# If the 'errors' output contains one or more error messages, Atmos considers the policy failed.
+
+errors["for the 'my-component' component, the variable 'name' must be provided on the command line using the '-var' flag"] {
+    not input.cli_vars.name
+}
+```
+</File>
+
+When executing the following command (and not passing the `name` variable on the command line), Atmos will validate
+the component using the OPA policy, which will fail and prevent the component from being provisioned:
+
+<Terminal>
+```console
+atmos terraform apply my-component -s plat-ue2-dev
+
+Validating the component 'my-component' using OPA file 'my-component/validate-my-component.rego'
+
+for the 'my-component' component, the variable 'name' must be provided on the command line using the '-var' flag
+```
+</Terminal>
+
+On the other hand, when passing the `name` variable on the command line using the `-var name=api` flag, the command will succeed:
+
+<Terminal>
+```shell
+atmos terraform apply my-component -s plat-ue2-dev -var name=api
+```
+</Terminal>
+
+### Restrict a Terraform variable from being provided on the command line
+
+If you need to prevent a Terraform variable from being passed (and overridden) on the command line,
+add the following OPA policy in the file `stacks/schemas/opa/my-component/validate-my-component.rego`
+
+<File title="stacks/schemas/opa/my-component/validate-my-component.rego">
+```rego
+package atmos
+
+errors["for the 'my-component' component, the variable 'name' cannot be overridden on the command line using the '-var' flag"] {
+    input.cli_vars.name
```

Comment:
I find CLI vars confusing because they are `-var` args right? So these are the ones passed after the double dash. So they should be specific to terraform.

Terraform uses the TF_CLI_ prefix, so I think we can use the same.
```suggestion
    input.tf_cli_vars.name
```